src/controllers/articleController.js

```diff
@@ -190,16 +189,82 @@ class articleManager {
           favoritesCount,
           author
         };
-        cacheArticles.push(art);
+        cachedArticles.push(articleInfo);
       }
       return res.status(200).json({
-        articles: cacheArticles
+        articles: cachedArticles
       });
     } catch (error) {
       return res.status(404).json({
         error: 'there is no article'
       });
     }
   }
+
+  /**
+   *
+   * @param {integer} id
+   * @param {string} slug
+   * @returns {string} acknowledgement message
+   */
+  static async likeArticle(id, slug) {
+    const likeA = {
+      userId: id,
+      slug,
+      like: true,
+      dislike: false
+    };
+    await likeDislikes.create(likeA);
+    return true;
+  }
+
+  /**
+   *
+   * @param {string} slug
+   * @returns {integer} number of likes
+   */
+  static async countLikesDislikes(slug) {
+    const likes = await likeDislikes.findAll({
+      where: { slug, like: true }
+    });
+    const disLikes = await likeDislikes.findAll({
+      where: { slug, dislike: true }
+    });
+    return ([likes.length, disLikes.length]);
+  }
+
+  /**
+ *
+ * @param {object} req
+ * @param {object} res
+ * @returns {object} liked or disliked article
+ */
+  static async likeDislike(req, res) {
```

Comment:
This function is doing a lot of things. You need to separate the different functionalities, so that it handles a single aspect.
Look at 'https://hackernoon.com/you-dont-understand-the-single-responsibility-principle-abfdd005b137'

Reply:
done